lib/tasks/sanitize_data.rake

```diff
@@ -0,0 +1,162 @@
+      address.address1 = address.address2 = "Dummy address"
+      address.city = "Dummy City"
+      address.zipcode = "0000"
+      address.company = address.latitude = address.longitude = nil
+      address.save!
+    end
+
+    def sanitize_payment_method(payment_method)
+      payment_method.name = "Payment Method #{payment_method.id}"
+      payment_method.description = payment_method.name
+      payment_method.save!
+    end
+
+    def sanitize_credit_card(card)
+      card.month = card.year = card.start_month = card.start_year = 1
+      card.cc_type = card.first_name = card.last_name = card.last_digits = "Dummy"
+      card.save!
+    end
+
+    def sanitize_payment(payment)
+      payment.response_code = payment.avs_response = payment.cvv_response_code = nil
+      payment.identifier = payment.cvv_response_message = nil
+      payment.save!
+    end
+
+    def sanitize_paypal_express_checkout(paypal_express_checkout)
+      paypal_express_checkout.token = nil
+      paypal_express_checkout.save!
+    end
+
+    def sanitize_stripe_account(stripe_account)
+      stripe_account.stripe_user_id = stripe_account.stripe_publishable_key = nil
+      stripe_account.save!
+    end
+
+    def sanitize_batch(model_class, sanitize_method_name, canned_users = nil)
+      model_class.constantize.all.each_with_index do |object, index|
+        sanitized_user = sanitized_user(canned_users, index) if canned_users
+
+        if sanitized_user
+          puts "Sanitizing #{model_class} #{object.id} with #{sanitized_user[:first_name]}"
+          __send__ sanitize_method_name, object, sanitized_user
+        else
+          puts "Sanitizing #{model_class} #{object.id}"
+          __send__ sanitize_method_name, object
+        end
+      end
+    end
+
+    def sanitized_user(canned_users, index)
+      sanitized_user = canned_users[index % canned_users.size]
+      sanitized_user[:email] = "#{index}_#{sanitized_user[:email]}"
+      sanitized_user
+    end
+
+    def build_canned_users
+      base_canned_users.each do |user|
+        user[:email] = "ofn@example.com"
+        user[:phone] = "0123456789"
+      end
+    end
+
+    def base_canned_users
+      [{ first_name: "Bob", last_name: "Jones" },
+       { first_name: "Cindy", last_name: "Rest" },
+       { first_name: "Pete", last_name: "Smith" },
+       { first_name: "Tony", last_name: "Ballantyne" },
+       { first_name: "Ben", last_name: "Raven" },
+       { first_name: "Robyn", last_name: "Monster" },
+       { first_name: "Nako", last_name: "Tolkein" },
+       { first_name: "Helen", last_name: "Mitcham" },
+       { first_name: "Emma", last_name: "Low" },
+       { first_name: "Mandy", last_name: "Trust" }]
+    end
+
+    # We need to disable user coNFirmation emails that would otherwise be triggered
+    def disable_user_confirmation_emails
+      Spree::UserMailer.class_eval do
+        def confirmation_instructions(*) end
+      end
```

Comment:
We really need to stop abusing class evil. What about disabling ActionMailer deliveries in this rake task? See perform_deliveries in https://guides.rubyonrails.org/v3.2/action_mailer_basics.html#action-mailer-configuration.

Reply:
yes, ActionMailer::Base.perform_deliveries = false would do the trick, I ended up switching to update_all that doesnt trigger the emails.